src/components/CSVButton/CSVButton.jsx

```diff
@@ -0,0 +1,258 @@
+        'Gender',
+        'Ethnicity',
+        'Student Group ID',
+        'Post-Test A',
+        'Post-Test A Notes',
+        'Post-Test R',
+        'Post-Test R Notes',
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    if (type === 'site') {
+      return [
+        'Site ID',
+        'Student ID',
+        'First Name',
+        'Last Name',
+        'Grade',
+        'Home Teacher',
+        'Gender',
+        'Ethnicity',
+        'Student Group ID',
+        'Post-Test A',
+        'Post-Test A Notes',
+        'Post-Test R',
+        'Post-Test R Notes',
+        'Pre-Test A',
+        'Pre-Test A Notes',
+        'Pre-Test R',
+        'Pre-Test R Notes',
+      ];
+    }
+    return [];
+  }
+
+  function mapResponse() {
+    if (type === 'allAreas') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'area') {
+      return studentResponseData.map(student => [
+        student.areaId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    if (type === 'site') {
+      return studentResponseData.map(student => [
+        student.siteId,
+        student.studentId,
+        student.firstName,
+        student.lastName,
+        student.grade,
+        student.homeTeacher,
+        student.gender,
+        student.ethnicity,
+        student.studentGroupId,
+        student.posttestA,
+        student.posttestANotes,
+        student.posttestR,
+        student.posttestRNotes,
+        student.pretestA,
+        student.pretestANotes,
+        student.pretestR,
+        student.pretestRNotes,
+      ]);
+    }
+    return [];
+  }
+
+  const CSVReport = {
+    data: studentResponseData,
+    headers: createHeaders(),
+    filename: 'Areas_Report.csv',
+  };
+
+  const addAssociatedSiteToArea = async resData => {
+    async function fetchStudentsByAllAreas() {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const element of resData) {
+        TLPBackend.get(`/students/area/${element.areaId}`)
```

Comment:
I don't understand this. If this function fetches all students, shouldn't we just make one request GET to `/students/`. That route includes `areaId` in its response. While this seems okay right now, I think it will take a really long time to produce the report once there are more students.